elastic/kibana
# [TSVB] Enables url drilldowns for range selection

## Summary

Closes https://github.com/elastic/kibana/issues/92164
Until now, TSVB supports only the brush event by using the `APPLY_FILTER` trigger directly. As a result, the user can't create url drilldowns as they require the `RANGE_SELECT_TRIGGER ` instead.

This PR changes the event type and also converts the data to DataTable in order for the event to work properly. The brush event actually requires only the timefilter to work correctly but as we also want to enable filter click on TSVB (currently not supported https://github.com/elastic/kibana/issues/83892) I have created the `convertSeriesToDataTable` function as a step towards this direction.

### How to test it
Go to Dashboard and create a timeseries TSVB visualization. On edit mode create a url drilldown (license should be updated first). You can see that now, there is an extra option `Range Selection`. You can use Discover as the `Go to drilldown`

```
{{kibanaUrl}}/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'{{date event.from}}',to:'{{date event.to}}'))&_a=(columns:!(_source),filters:{{rison context.panel.filters}},index:'{{context.panel.indexPatternId}}',interval:auto,query:(language:{{context.panel.query.language}},query:'{{context.panel.query.query}}'),sort:!())
```

Save the drilldown, switch to view mode and then brush on the TSVB vis. You should see a popup with the url drilldown you have just created as an option. Click it, it should navigate to Discover with the correct timeframe.

![image](https://user-images.githubusercontent.com/17003240/112950897-7794aa00-9143-11eb-9a17-6be438460e44.png)


### Checklist

Delete any items that are not applicable to this PR.
